test/e2e/proxy_test.go

```diff
@@ -61,10 +65,126 @@ func proxyTestSuite(t *testing.T) {
 	// the deletion of TrustedCAConfigMap, which the former relies on
 	t.Run("Certificate validation", tc.testCertsImport)
 	t.Run("Environment variables validation", tc.testEnvVars)
+
+	t.Run("External requests respect global proxy", tc.testProxyRequests)
+
 	t.Run("Certificate validation removal", tc.testCertsRemoval)
 	t.Run("Environment variables removal validation", tc.testEnvVarRemoval)
 }
 
+// testProxyRequest tests that external requests from each Windows node properly go through the proxy
+// and circumvent the proxy if the target address is specified in the NO_PROXY list
+func (tc *testContext) testProxyRequests(t *testing.T) {
+	clusterProxy, err := tc.client.Config.ConfigV1().Proxies().Get(context.TODO(), "cluster", meta.GetOptions{})
+	require.NoError(t, err)
+
+	// variables to look for in the curl output
+	httpVal := fmt.Sprintf("http_proxy == '%s'", clusterProxy.Status.HTTPProxy)
+	noProxyVal := fmt.Sprintf("no_proxy == '%s'", clusterProxy.Status.NoProxy)
+
+	for _, node := range gc.allNodes() {
+		t.Run(node.GetName(), func(t *testing.T) {
+			addr, err := controllers.GetAddress(node.Status.Addresses)
+			require.NoError(t, err, "unable to get node address")
+
+			t.Run("HTTP_PROXY request", func(t *testing.T) {
+				httpUrl := "http://raw.githubusercontent.com/openshift/windows-machine-config-operator/master/README.md"
+				out, err := tc.curlHeadersOnly(addr, httpUrl)
+				require.NoErrorf(t, err, "unable to get curl response to URL %s", httpUrl)
+
+				httpVarInUse, noProxyVarInUse, proxyInViaHeader := verifyCurlOutput(out, httpVal, noProxyVal)
+
+				assert.Truef(t, httpVarInUse, "expected http_proxy %s in request", clusterProxy.Status.HTTPProxy)
+				assert.Truef(t, noProxyVarInUse, "expected no_proxy %s in request", clusterProxy.Status.NoProxy)
+				assert.True(t, proxyInViaHeader, "'Via' header with proxy expected for proxied request")
+			})
+
+			t.Run("NO_PROXY request", func(t *testing.T) {
+				// using an endpoint that appears in the noProxyList const
+				nonProxiedUrl := "google.com"
```

Comment:
Please derive this from the noProxyList variable.
As it is, if we want to change the URL it needs to be changed in two separate spots and thats not clear.